### DISCUSSION TITLE
联系表单：关于后台功能相关的可行性

### DISCUSSION THREAD

**Discussion by w-flac** [halo-dev/halo#5895] [General]
### Your current Halo version

2.15.2

### Describe this feature

使用痛点：
目前的联系表单插件存在诸多不便，即使在配置了邮件推送后，获取表单数据多为组合表单，没办法对现有的数据做到直接回复，如需回复，需要单独复制邮箱地址，但这样的问题是没办法引用对方的表单数据。

期望功能：
1、后台集成表单的回复：对表单数据进行后台的直接引用，通过配置的邮件服务直接回复。
2、联系表单的自定义样式：希望后台能够集成对于表单的自定义样式，现在的表单样式真的比较丑，一整个表格不是很 ok

**Comment by LIlGG** [comment]
不太理解你所描述的表单回复的需求是什么样的，表单通常可以匿名收集信息，因而可能无法进行回复。

由于此问题无法在 Halo 中进行解决，我将转移至 Discussion ，应用市场插件的问题建议前往该应用讨论区中进行。

**Reply by w-flac** [reply]
单从匿名收集信息的话，这个插件无疑是成功的，在目前没有工单系统的情况下，没办法对已有的表单进行引用回复。

我来列举一下演示场景：

比如：当我创建表单后，用户提交了数据，插件后端获取了内容

![image](https://github.com/halo-dev/halo/assets/154269470/3aae4dc5-332f-42dc-b286-fc0e369787bd)

配置了邮件服务之后，邮箱会直接发送到公共邮箱组，由企业邮箱进行分发至项目组，而负责这个项目的成员会针对收集的数据进行工程的修改，（没错，我们是通过邮件进行客户沟通的），当提交的问题修改完毕后，以往我们是直接复制表单内容到邮件里，很麻烦，但这也导致多了很多可减少的操作步骤。

相反，如果在后台可以回复的话，直接引用提交的表单数据，可以减少不必要的操作步骤。

现阶段如果没有工单系统的话，只能通过表单插件进行信息的收集及回馈

**Reply by LIlGG** [reply]
所以实际上这个问题属于特殊性而不是通用性问题。

本质在于将联系表单当成了工单系统，这与之前有人将联系表单用于友链提交一样，他也想让联系表单自动添加提交的友链至友联插件。

因此我个人认为不应当由联系表单来支持这种功能，联系表单应该只是一个单纯的收集、展示信息的产品，至于收集到信息如何使用已经超出了此插件的范围。

不过有一种解决方式，联系表单中可以加入 hook。在触发提交 hook 时，由用户决定如何处理提交数据。但这个功能必要性还有待讨论，可行性也有待研究。
